values-hub.yaml

```diff
@@ -72,6 +72,18 @@ clusterGroup:
       timeout: 234
       # verbosity: "-v"
 
+  # Explicitly mention the cluster-state based overrides we plan to use for this pattern.
+  # We can use self-referential variables because the chart calls the tpl function with these variables defined
+  sharedValueFiles:
+    - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}.yaml'
+
+  # To mirror the "Classic" magic include structure, the clusterGroup would need all of these:
+  # sharedValueFiles:
+  #   - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}.yaml'
+  #   - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}-{{ $.Values.global.clusterVersion }}.yaml'
+  #   - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}-{{ $.Values.clusterGroup.name }}.yaml'
+  #   - '/cluster-overrides/values-{{ $.Values.global.clusterVersion }}-{{ $.Values.clusterGroup.name }}.yaml"
```

Comment:
Mainly for consistency reasons it might make sense to add the localClusterName commented out example here too?

Reply:
Yeah. localClusterName would make sense here.

Since I can't find a good place to comment on the ignoreMissingFiles question otherwise: I think that needs to stay, and should stay. The reason for that is that if we want a default to "fall through" or not define an exception for one platform or another, the lack of a file will simply be a fall-through case.

This, for example, if we want an override for All 4.12 clusters, only only AWS on 4.13, we could do:
```yaml
- '/cluster-overrides/values-{{ $.Values.global.clusterVersion }}.yaml'
- '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}-{{ $.Values.global.clusterVersion }}.yaml'
```

And then the versions that we don't want to have exceptions for would look for those files, find nothing, not error, and proceed (as we'd want them to, I think). Meanwhile, the combinations that do require exceptions would process the named files for them and also have the desired behavior.

Reply:
I'm inclined to think that we should (eventually) remove the implicit magic variables entirely, but I don't have a good sense for how much others might be using them.

The only downside I see to the new version is that each clusterGroup has to declare its own overrides in this way, which could lead to some duplication if they clustergroups are using the same charts. HOWEVER, I think an argument can be made that explciit is better than implicit in this situation, and the flexibility of the templated system outweighs the duplication problem.

I do think that supporting both schemes forever would not be a great idea, as it might be possible for unintended consequences to occur with files in "magic" format but not explicit, or something like that. I don't at present have a strong sense for how much dependence there is on this scheme; my gut feeling is to remove them sooner than later, so that consumers and other pattern developers can learn better habits.

Reply:
Once we start using clusterGroup as a chart, and can create a new major version, that would be a good time to drop the magic (and make other backed up breaking changes)